lib/modules/manager/cargo/extract.ts

```diff
@@ -43,7 +53,15 @@ function extractFromSection(
         currentValue = version;
         nestedVersion = true;
         if (registryName) {
-          const registryUrl = cargoRegistries[registryName];
+          let registryUrl = cargoRegistries[registryName];
+
+          if (!registryUrl) {
+            const envUrl = get_cargo_index_env(registryName);
+            if (envUrl) {
+              registryUrl = envUrl;
+            }
+          }
```

Comment:
```suggestion
          const registryUrl = cargoRegistries[registryName] ?? getCargoIndexEnv(registryName);
```
Same as above and nested if clause has no effect.